salt/metalk8s/orchestrate/deploy_new_node.sls

```diff
@@ -1,47 +1,61 @@
-{%- if pillar['bootstrap_id'] %}
-{%-   set control_plane_ips = salt.saltutil.runner('mine.get', tgt=pillar['bootstrap_id'], fun='control_plane_ip') %}
-{%- else %}
-{%-   set control_plane_ips = {} %}
-{%- endif %}
+{%- set control_plane_ip = salt.saltutil.runner(
+    'mine.get',
+    tgt=pillar['bootstrap_id'],
+    fun='control_plane_ip'
+)[pillar['bootstrap_id']] %}
 
-{%- if pillar['bootstrap_id'] in control_plane_ips.keys() and control_plane_ips[pillar['bootstrap_id']] %}
-{%-   set control_plane_ip = control_plane_ips[pillar['bootstrap_id']] %}
+{%- set pillar_data = {
+    'repo': {
+        'host': control_plane_ip
+    },
+    'registry_ip': control_plane_ip
+} %}
+
+{%- if 'metalk8s' in pillar
```

Comment:
No. The `version` must always be set on the `Node` object. If it's not, that's a failure. No fallback.